### PR TITLE
SUIT-13423 - Config variables cannot be used and saved in non-text fi…

### PR DESCRIPTION
**Pull request by AntonSBilan**
allow st config vars in chains

### CODE REVIEW

File: lib/utils/socketChainHelper.js [hunk 1]
```diff
@@ -157,8 +157,13 @@ const applyCountAndDelayAndClicks = (socketMessage, data) => {
  * @param data
  * @returns {number}
```
Comment (givehug) [inline lib/utils/socketChainHelper.js]: update this @returns {number}

File: lib/validation/validators.js [hunk 2]
```diff
@@ -131,6 +131,18 @@ const validatePositiveNumber = (val, name) => {
 	return val;
 };
 
+const validateStVarOrPositiveNumber = (val, name) => {
+	if (!['number', 'string'].includes(typeof val)) {
+		throwError(name + 'should be suitest configuration variable or number');
```
Comment (givehug) [inline lib/validation/validators.js]: missing space 'should be...'

File: lib/validation/validators.js [hunk 3]
```diff
@@ -131,6 +131,18 @@ const validatePositiveNumber = (val, name) => {
...
+	} else if (typeof val === 'number' && !Number.isFinite(val) || val < 0) {
+		throwError(name + ' should be positive number');
+	} else if (typeof val === 'string' && !/(<%.+%>)/.test(val)) {
+		throwError(name + 'should be suitest configuration variable');
```
Comment (givehug) [inline lib/validation/validators.js]: missing space 'should be...'

File: lib/validation/validators.js [hunk 4]
```diff
@@ -131,6 +131,18 @@ const validatePositiveNumber = (val, name) => {
 	return val;
 };
 
+const validateStVarOrPositiveNumber = (val, name) => {
+	if (!['number', 'string'].includes(typeof val)) {
+		throwError(name + 'should be suitest configuration variable or number');
+	} else if (typeof val === 'number' && !Number.isFinite(val) || val < 0) {
+		throwError(name + ' should be positive number');
```
Comment (givehug) [inline lib/validation/validators.js]: you can use validatePositiveNumber() here

File: lib/validation/validators.js [hunk 5]
```diff
@@ -131,6 +131,18 @@ const validatePositiveNumber = (val, name) => {
 	return val;
 };
 
+const validateStVarOrPositiveNumber = (val, name) => {
```
Comment (givehug) [inline lib/validation/validators.js]: please add UT for this function

File: lib/constants/validationKeys.js [hunk 6]
```diff
@@ -3,6 +3,7 @@ const validationKeys = {
 	CONFIGURE: Symbol('configure'),
 	NON_EMPTY_STRING: Symbol('nonEmptyString'),
 	POSITIVE_NUMBER: Symbol('positiveNumber'),
+	ST_VAR_OR_POSITIVE_NUMBER: Symbol('nonStVarOrPositiveNumber'),
```
Comment (JeKachan) [inline lib/constants/validationKeys.js]: "non" prefix is redundant I guess?

File: testDefinition/examplesTS/sleepChain.ts [hunk 7]
```diff
@@ -1,7 +1,7 @@
 import * as suitest from '../../index';
 
 async function sleepTest() {
-	const sampleSleep = suitest.sleep(10);
+	const sampleSleep = suitest.sleep(null);
```
Comment (JeKachan) [inline testDefinition/examplesTS/sleepChain.ts]: sleep can accept null?

Comment (AntonSBilan) [reply testDefinition/examplesTS/sleepChain.ts]: on fe it is so, one can set null by `.` to sleep line

**Pull request merged**